src/EnableSharedFromThis.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <memory>
+
+namespace reactivesocket {
+
+class EnableSharedFromThisVirtualBase
+    : public std::enable_shared_from_this<EnableSharedFromThisVirtualBase> {};
+
+template <typename T>
+class EnableSharedFromThisBase
```

Comment:
I want user to be able to implement a class which inherits both from SubscriberBase and SubscriptionBase. The class needs to inherit std::enable_shared_from_this only once in order to work properly. Unfortunately std::enable_shared_from_this can't be inherited virtually because of some implementation details of that class.
EnableSharedFromThis is essentially a workaround which will enable you to inherit std::enable_shared_from_this only once in your class hierarchy.